packages/mail-translation/src/generateRazorOutputTemplates.ts

```diff
@@ -0,0 +1,63 @@
+import { TranslatedTemplate } from "./processTemplate"
+
+export function generateRazorOutputTemplates({
+  translatedTemplates,
+  defaultLanguage,
+}: {
+  translatedTemplates: TranslatedTemplate[]
+  defaultLanguage?: string
+}) {
+  return translatedTemplates.map(template => ({
+    filename: getFilename({ template, defaultLanguage }),
+    content: escapeRazorConflicts(template.content),
+  }))
+}
+
+function escapeRazorConflicts(templateContent: string): string {
+  const cssAtRules = [
+    "annotation",
+    "character-variant",
+    "charset",
+    "color-profile",
+    "container",
+    "counter-style",
+    "font-face",
+    "font-feature-values",
+    "font-palette-values",
+    "import",
+    "keyframes",
+    "-webkit-keyframes",
+    "layer",
+    "media",
+    "namespace",
+    "ornaments",
+    "page",
+    "position-try",
+    "property",
+    "scope",
+    "starting-style",
+    "styleset",
+    "stylistic",
+    "supports",
+    "swash",
+    "view-transition",
+  ].join("|")
+
+  return templateContent.replace(new RegExp(`(?<!@)@(${cssAtRules})`, "g"), "@@$1")
+}
+
+function getFilename({
+  template,
+  defaultLanguage,
+}: {
+  template: TranslatedTemplate
+  defaultLanguage?: string
+}): string {
+  if (template.language === defaultLanguage || !defaultLanguage) {
+    return template.isPlaintext ? `${template.name}.txt.cshtml` : `${template.name}.cshtml`
+  } else {
+    return template.isPlaintext
+      ? `${template.name}.${template.language}.txt.cshtml`
+      : `${template.name}.${template.language}.cshtml`
+  }
```

Comment:
Hmm, there is a lot of repeated code maybe we can do smth like that

```suggestion
  const includeLanguage = !!defaultLanguage && template.language !== defaultLanguage 

  return [template.name, 
    ...(includeLanguage ? [template.language] : []),
    ...(isPlainText ? ["txt"]: []), 
    "cshtml"].join(".")

WDYT?

Reply:
Sure, but we will have to live with adjusting 4 filenames instead of just one :sweat_smile:

Reply:
I think it's very unlikely that we will have to ever touch this function, so I think it's better to prefer readability in such case. However, I would say that both solutions seem valid, so if you insist on having less repetitions I'll change that 😅